### PR TITLE
[3.6]don't attempt to client-bind Broadcast Bind addresses (#12873)

### PR DESCRIPTION
**Pull request by dothebart**
backport of https://github.com/arangodb/arangodb/pull/12873

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12388/

**Comment by dothebart** [comment]
tests blue except for known garbage collection issue in `server_permissions`.

**Review by jsteemann — APPROVED** [review]

File: arangosh/Shell/ClientFeature.cpp [hunk 1]
```diff
@@ -234,6 +234,25 @@ void ClientFeature::validateOptions(std::shared_ptr<ProgramOptions> options) {
     FATAL_ERROR_EXIT();
   }
 
+  if (!_endpoint.empty() &&
+      (_endpoint != "none") &&
+      (_endpoint != Endpoint::defaultEndpoint(Endpoint::TransportType::HTTP))) {
+    std::unique_ptr<Endpoint> endpoint(Endpoint::clientFactory(_endpoint));
+    if (endpoint.get() == nullptr) {
```
Comment (jsteemann) [inline arangosh/Shell/ClientFeature.cpp:241]: apparently this breaks VST client connections, so we should change this first before we merge it.

**Review by jsteemann — CHANGES_REQUESTED** [review]
This change will break VST connections via the arangosh.

File: arangosh/Shell/ClientFeature.cpp (hunk 1 above)
Comment (jsteemann) [reply arangosh/Shell/ClientFeature.cpp]: Fixed

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12484/

**Review by jsteemann — APPROVED** [review]

**Review by dothebart — COMMENTED** [review]
lgtm

**Comment by KVS85** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12534/

**Comment by KVS85** [comment]
Tests blue.

**Pull request merged**